test/test_profiler.py

```diff
@@ -1285,6 +1285,62 @@ def test_tensor_properties(self):
         device_info = [x.device if x else None for x in input_info.tensor_metadata]
         self.assertEqual(device_info, [torch.device("cpu"), torch.device("cpu"), None])
 
+    def test_sparse_tensors(self):
+        i = [[0, 1, 1], [2, 0, 2]]
+        v = [3, 4, 5]
+        s = torch.sparse_coo_tensor(i, v, (2, 3))
+
+        with profile(with_stack=True, profile_memory=True, record_shapes=True) as p:
+            _ = s + s
+
+        nodes = p.profiler.kineto_results.experimental_event_tree()
+        node = find_node_with_name(nodes, "aten::add")
+        self.assertIsNotNone(node)
+
+        self.assertIsInstance(
+            node.extra_fields,
+            torch._C._profiler._ExtraFields_TorchOp)
+
+        self.assertEqual(node.extra_fields.inputs.shapes, [[2, 3], [2, 3], []])
+        self.assertEqual(node.extra_fields.inputs.strides, [[], [], []])
+
+        input_info = node.extra_fields.inputs
+
+        # FIXME: Different systems have different names for int64_t
+        # self.assertEqual(input_info.dtypes, ['long long', 'long long', 'Scalar'])
+        self.assertIn(input_info.dtypes[0], ["long long", "long int", "long"])
```

Comment:
@robieta It likely makes sense change the linked line below to be "Tensor" instead given that the metadata already has the dtype of the tensor, and that string this gives for an int64_t tensor is system dependent.

However, that is outside the scope of this diff.
https://github.com/pytorch/pytorch/blob/09e837634bc76c4380db7119e0e997816d584844/torch/csrc/profiler/collection.cpp#L111